modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/quantity_selector/QuantitySelector.js

```diff
@@ -12,246 +12,122 @@
-	);
-	const [prevAvailable, setPrevAvailable] = useState(
-		currentQuantity - props.multipleQuantity >= props.minQuantity
-	);
-	const [isThrottling, setIsThrottling] = useState(false);
+	const [selectedQuantity, setSelectedQuantity] = useState(props.quantity);
+
+	const quantitiesList = generateOptions(props.settings);
+
+	const updateQuantity = (q) => {
+		if (q !== null) {
+			setSelectedQuantity(q);
+			try {
+				props.updateQuantity(q);
+			}
+			catch (e) {
+				// eslint-disable-next-line no-console
+				console.log('stand alone component');
+			}
+		}
+	};
 
-	const inputRef = createRef();
+	const inputStyle = getInputStyle(props.inputSize);
 
 	useEffect(() => {
-		setCurrentQuantity(props.quantity);
-	}, [props.quantity, setCurrentQuantity]);
-
-	useEffect(() => {
-		if (props.throttleOnUpdate) {
-			setIsThrottling(true);
-
-			props.onUpdate(currentQuantity).then(() => setIsThrottling(false));
-		}
-		else {
-			props.onUpdate(currentQuantity);
-		}
+		// eslint-disable-next-line no-unused-expressions
+		props.quantity
+			? updateQuantity(props.quantity)
+			: updateQuantity(selectedQuantity);
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [currentQuantity]);
-
-	useEffect(() => {
-		setNextAvailable(
-			currentQuantity + props.multipleQuantity <= props.maxQuantity
-		);
-		setPrevAvailable(
-			currentQuantity - props.multipleQuantity >= props.minQuantity
-		);
-	}, [
-		currentQuantity,
-		props.maxQuantity,
-		props.minQuantity,
-		props.multipleQuantity,
-	]);
-
-	function updateCurrentQuantity(newQuantity) {
-		if (
-			newQuantity >= props.minQuantity &&
-			newQuantity <= props.maxQuantity &&
-			newQuantity % props.multipleQuantity === 0
-		) {
-			setCurrentQuantity(newQuantity);
-		}
-	}
-
-	function _increaseQuantity() {
-		if (nextAvailable) {
-			updateCurrentQuantity(currentQuantity + props.multipleQuantity);
-		}
-	}
-
-	function _decreaseQuantity() {
-		if (prevAvailable) {
-			updateCurrentQuantity(currentQuantity - props.multipleQuantity);
-		}
-	}
-
-	function handleInputChange() {
-		const {value} = inputRef.current;
-
-		return updateCurrentQuantity(parseInt(value, 10));
-	}
-
-	const decreaseQuantity = throttle(_decreaseQuantity, THROTTLE_TIMEOUT),
-		increaseQuantity = throttle(_increaseQuantity, THROTTLE_TIMEOUT);
-
-	function handleInputKeyUp(e) {
-		switch (e.key) {
-			case 'ArrowUp':
-				increaseQuantity();
-				break;
-			case 'ArrowDown':
-				decreaseQuantity();
-				break;
-			case 'Enter':
-			default:
-				break;
-		}
-	}
-
-	function handleSelectChange() {
-		const {value} = inputRef.current;
-
-		setCurrentQuantity(value);
-	}
-
-	let btnSizeClass;
-	let formControlSizeClass;
-
-	if (props.size === 'large') {
-		btnSizeClass = 'btn-lg';
-		formControlSizeClass = 'form-control-lg';
-	}
-
-	if (props.size === 'small') {
-		btnSizeClass = 'btn-sm';
-		formControlSizeClass = 'form-control-sm';
-	}
+	}, [props, selectedQuantity, setSelectedQuantity, props.quantity]);
```

Comment:
Here again it feels like we're using lint suppression too eagerly. For example, why is `setSelectedQuantity` in the list here (it's from React `useState` so it should be stable)? Would we have noticed it if we hadn't disabled the lint?